pkg/yamltemplate/filetests/ytt-library/json.yml

```diff
@@ -7,24 +7,51 @@ fragment:
   piece1: false
 #@ end
 
-test1: #@ json.encode({"a": [1,2,3,{"c":456}], "b": "str"})
-test1a: #@ json.encode(yaml_fragment())
-test1b: #@ json.encode({"inside_map": yaml_fragment(), "inside_array": [yaml_fragment()]})
-test2: #@ json.encode({})
-test3: #@ json.decode("{}")
-test4: #@ json.decode('{"a":[1,2,3,{"c":456}],"b":"str"}')
+encode:
+  test1: #@ json.encode({})
+  test2: #@ json.encode({"a": [1,2,3,{"c":456}], "b": "str"})
+  test3: #@ json.encode(yaml_fragment())
+  test4: #@ json.encode({"inside_map": yaml_fragment(), "inside_array": [yaml_fragment()]})
+  indent:
+    test1: #@ json.encode({"a": [1,2,3,{"c":456}], "b": "str"}, indent=4)
+    test2: #@ json.encode({"a": [1,2,3,{"c":456}], "b": "str"}, indent=0)
+  #! TODO: how should we test error cases like these?
+  #! test1e: #@ json.encode({"a": [1,2,3,{"c":456}], "b": "str"}, indent=5)
+  #! test1f: #@ json.encode({"a": [1,2,3,{"c":456}], "b": "str"}, inschment=4)
+  #!
+  #! TODO: update docs to include indent:
+  #! https://github.com/vmware-tanzu/carvel.dev/blob/develop/content/ytt/docs/latest/lang-ref-ytt.md#serialization-modules
```

Comment:
Let's delete. Moved this convo to the issue.

Reply:
Deleted